chain/client/src/sync/external.rs

```diff
@@ -201,8 +216,22 @@ impl ExternalConnection {
     }
 }
 
-/// Construct a location on the external storage.
-pub fn external_storage_location(
+/// Construct the header location on the external storage.
+pub fn external_header_storage_location(
```

Comment:
nit: you can refactor to only have one function e.g. `external_file_storage_location` that takes the type as param. Same for the header_filename and part_filename assuming you apply Longarithm's suggestion.